### PR TITLE
Add support for cathash image creation in android client

### PR DESCRIPTION
**Pull request by HenrikJannsen**
Based on https://github.com/bisq-network/bisq-mobile/pull/86

We have to build the cathash image on the client as otherwise the loading of the images would take too long. We cache the images once created so they are only generated once. 
Beside that to provide the image from the Rest API in headless mode would have been quite a hack (I got it working but did not like it - required to start JavaFX in headless mode via the monocle test framework which supports that).

I tried to add iOS support but did not get the images displayed. Beside that the iOS code is ChatGPT derived and must be reviewed/improved/fixed by a dev familiar with iOS.

### CODE REVIEW

**Comment by HenrikJannsen** [comment]
Continue working on ios.... got the loading of the images resolved. Still having problems to display it...

**Comment by rodvar** [comment]
henrik still working on this one, marking as draft

**Comment by rodvar** [comment]
@HenrikJannsen Haven't been able to work this properly in the `xClients` so decided to make my way through it understanding the node part better. I integrate your work with the presentation architecture so now the final generated cathash is accessible to the rest of the app components via a repository.

It works like charm on androidNode, I've did all the changes considering android and iOS and fix some issues and compilation along the way so later today when I get back to the `xClients` hopefully the way with be paved to success :) 

I'll let you know, I can smell the merge coming soon... please have a look at my changes if you are around 💪

File: shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/components/molecules/TopBarPresenter.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import network.bisq.mobile.domain.PlatformImage
+import network.bisq.mobile.domain.data.repository.UserRepository
+import network.bisq.mobile.presentation.BasePresenter
+import network.bisq.mobile.presentation.MainPresenter
+
+open class TopBarPresenter(
```
Comment (rodvar) [inline shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/components/molecules/TopBarPresenter.kt:11]: @nostrbuddha had to add a TopBarPresenter in this PR hope that is ok, happy to help if you get conflicts.

**Comment by HenrikJannsen** [comment]
Ah great that you could get it working on iOS!!! And added support for the top bar!

File: shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/components/molecules/TopBarPresenter.kt (hunk 1 above)
Comment (nostrbuddha) [reply shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/components/molecules/TopBarPresenter.kt:11]: Sure. Rebased the UI code against this and making the UI PR now.

**Pull request merged**